docs/source/java/flight.rst

```diff
@@ -0,0 +1,215 @@
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient client = FlightClient.builder(allocator, location).build()){
+        // ... Consume operations exposed by Flight server
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+Cancellation and Timeouts
+=========================
+
+When making a call, clients can optionally provide `CallOptions`. This allows
+clients to set a timeout on calls. Also, some objects returned by client RPC calls
+expose a cancel method which allows terminating a call early.
+
+.. code-block:: Java
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+
+        Iterator<Result> resultIterator = tutorialFlightClient.doAction(
+                new Action("test-timeout"),
+                CallOptions.timeout(2, TimeUnit.SECONDS)
+        );
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+On the server side, timeouts are transparent. For cancellation, the server needs to manually poll
+``setOnCancelHandler`` or ``isCancelled`` to check if the client has cancelled the call,
+and if so, break out of any processing the server is currently doing.
+
+.. code-block:: Java
+
+    // Client
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+        try(FlightStream flightStream = flightClient.getStream(new Ticket(new byte[]{}))) {
+            // ...
+            flightStream.cancel("tutorial-cancel", new Exception("Testing cancellation option!"));
+        }
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+    // Server
+    @Override
+    public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        // ...
+        listener.setOnCancelHandler(()->{
+                    // Implement logic to handle cancellation option
+                });
+    }
+
+Enabling TLS
+============
+
+TLS can be enabled when setting up a server by providing a
+certificate and key pair to ``FlightServer.Builder.useTls``.
+
+On the client side, use ``Location.forGrpcTls`` to create the Location for the client.
+
+Enabling Authentication
+=======================
+
+.. warning:: Authentication is insecure without enabling TLS.
+
+Handshake-based authentication can be enabled by implementing
+on the client side through authenticates against the Flight service
+``FlightClient.authenticate``, or by authenticates with a username
+and password through ``FlightClient.authenticateBasic`` and
+``FlightClient.authenticateBasicToken``. And on the server side
+through ``FlightServer.Builder().authHandler`` or by
+``FlightServer.Builder().headerAuthenticator``.
```

Comment:
This mixes up several different things. Can we just stick to the wording already shown? We can add the descriptions of the other auth methods later and frankly they are all not very usable.

Reply:
Updated